cli/src/commands/extensions/extension.rs

```diff
@@ -207,11 +207,7 @@ pub fn validate_name(name: &str) -> Result<(), anyhow::Error> {
     if EXTENSION_NAME_RE.is_match(name) {
         Ok(())
     } else {
-        Err(anyhow!(
-            "{}: invalid extension name, must be lowercase alphanumeric, dash (-) or underscore \
-             (_)",
-            name
-        ))
+        Err(anyhow!("{}: invalid extension name, must be lowercase alphanumeric, dash (-)", name))
```

Comment:
I'd slightly reword, including the "starts alphabetic":
```suggestion
        Err(anyhow!("{}: invalid extension name, must start with a letter and can contain only lowercase alphanumeric character or dashes (-)", name))
```